src/main/java/dp/api/dataset/DatasetAPIClient.java

```diff
@@ -0,0 +1,378 @@
+        }
+    }
+
+    private void validate200ResponseCode(HttpRequestBase httpRequest, CloseableHttpResponse response) throws DatasetNotFoundException, UnexpectedResponseException {
+
+        switch (response.getStatusLine().getStatusCode()) {
+            case HttpStatus.SC_OK:
+                return;
+            case HttpStatus.SC_NOT_FOUND:
+                throw new DatasetNotFoundException(formatErrResponse(httpRequest, response));
+            default:
+                throw new UnexpectedResponseException(
+                        formatErrResponse(httpRequest, response),
+                        response.getStatusLine().getStatusCode());
+        }
+    }
+
+    private void addBody(Object object, HttpEntityEnclosingRequestBase httpRequest) throws JsonProcessingException, UnsupportedEncodingException {
+
+        String body = json.writeValueAsString(object);
+        StringEntity stringEntity = new StringEntity(body);
+        httpRequest.setEntity(stringEntity);
+    }
+
+    private void validateDatasetID(String datasetID) {
+        Args.check(isNotEmpty(datasetID), "a dataset id must be provided.");
+    }
+
+    private void validateEdition(String edition) {
+        Args.check(isNotEmpty(edition), "an edition must be provided.");
+    }
+
+    private void validateVersion(String version) {
+        Args.check(isNotEmpty(version), "a version must be provided.");
+    }
+
+    private void validateInstanceID(String instanceID) {
+        Args.check(isNotEmpty(instanceID), "a instance id must be provided.");
+    }
+
+    private static boolean isNotEmpty(String str) {
+        return str != null && str.length() > 0;
+    }
+
+    private void logRequest(HttpRequestBase httpRequest) {
+
+        new LogBuilder("Calling dataset API")
+                .addParameter("method", httpRequest.getMethod())
+                .addParameter("uri", httpRequest.getURI())
+                .log();
+
+    }
+
+    private void logResponse(HttpRequestBase httpRequest, CloseableHttpResponse response) {
+
+        new LogBuilder("dataset api response")
+                .addParameter("uri", httpRequest.getURI())
+                .addParameter("method", httpRequest.getMethod())
+                .addParameter("status", response.getStatusLine())
+                .log();
+    }
+
+    private <T> T parseResponseBody(CloseableHttpResponse response, Class<T> type) throws IOException {
+        HttpEntity entity = response.getEntity();
+        String responseString = EntityUtils.toString(entity);
+        return json.readValue(responseString, type);
+    }
+
+    private String formatErrResponse(HttpRequestBase httpRequest, CloseableHttpResponse response) {
+
+        return String.format("the dataset api returned a %s response for %s",
+                response.getStatusLine().getStatusCode(),
+                httpRequest.getURI());
+    }
+
+    public static void main(String[] args) throws Exception {
```

Comment:
Can the main method be moved into a Example class? As anything that imports the client will also take this main.

Reply:
Removed main method